tests/system/python/packages/test_authentication.py

```diff
@@ -0,0 +1,2211 @@
+        assert False, "reset package script failed!"
+
+    # Wait for foglamp server to start
+    time.sleep(wait_time)
+
+
+@pytest.fixture
+def remove_add_foglamp_pkgs():
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package && ./remove"
+                       .format(PROJECT_ROOT)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "remove package script failed!"
+
+    try:
+        subprocess.run(["cd {}/tests/system/python/scripts/package/ && ./setup {}"
+                       .format(PROJECT_ROOT, package_build_version)], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "setup package script failed"
+
+    try:
+        subprocess.run(["sudo apt install -y foglamp-south-http-south"], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "installation of packages failed"
+
+
+@pytest.fixture
+def enable_tls():
+    def _enable_tls(foglamp_url, wait_time, auth):
+        """
+        :param foglamp_url:
+        :param auth: None or password or certificate
+        """
+
+        if auth == 'password':
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/category/rest_api', json.dumps({"enableHttp": "false"}),
+                         headers={"authorization": PASSWORD_TOKEN})
+            r = conn.getresponse()
+            assert 200 == r.status
+
+            # FIXME: Remove this wait time
+            time.sleep(wait_time)
+
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/restart', json.dumps({}), headers={"authorization": PASSWORD_TOKEN})
+            r = conn.getresponse()
+            assert 200 == r.status
+
+        elif auth == 'certificate':
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/category/rest_api', json.dumps({"enableHttp": "false"}),
+                         headers={"authorization": CERT_TOKEN})
+            r = conn.getresponse()
+            assert 200 == r.status
+
+            # FIXME: Remove this wait time
+            time.sleep(wait_time)
+
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/restart', json.dumps({}), headers={"authorization": CERT_TOKEN})
+            r = conn.getresponse()
+            assert 200 == r.status
+        else:
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/category/rest_api', json.dumps({"enableHttp": "false"}))
+            r = conn.getresponse()
+            assert 200 == r.status
+
+            # FIXME: Remove this wait time
+            time.sleep(wait_time)
+
+            conn = http.client.HTTPConnection(foglamp_url)
+            conn.request("PUT", '/foglamp/restart', json.dumps({}))
+            r = conn.getresponse()
+            assert 200 == r.status
```

Comment:
I would suggest to reduce some line:

    headers = None
    if auth == 'password':
         headers={"authorization": TOKEN}
    elif auth == 'certificate':
          headers={"authorization": CERT_TOKEN}

and then call with headers every-time and remove repeated code. 

Note: May be we should only keep 1 global variable i.e. AUTH_TOKEN, do we have any test using TOKEN (with password based login) and CERT_TOKEN (with cert based login) at the same time?!

Reply:
Yes, in case of authmethod = any we have tests with both the tokens.